packages/sdk/src/oracle/factory.ts

```diff
@@ -0,0 +1,28 @@
+import assert from "assert";
+import { SortedRequests } from "./services/sortedRequests";
+import { ClientTable } from "./types/interfaces";
+import { State, OracleType, PartialConfigTable } from "./types/state";
+import { Emit } from "./store";
+import SkinnyFactory from "./skinnyFactory";
+import OptimisticFactory from "./optimisticFactory";
+
+export type PublicEmit = (oracleType: OracleType, state: State, prev: State) => void;
+const EventHandler = (oracleType: OracleType, publicEmit: PublicEmit): Emit => (state: State, prev: State) =>
+  publicEmit(oracleType, state, prev);
+
+export default (configTable: PartialConfigTable, emit: PublicEmit): ClientTable => {
```

Comment:
the factory now requires you key the config by the oracle type. you can only create 1 instance of each oracle type from the factory.  This is because the combined view of all requests ( across all chains and all oracles) need a way to locate the correct client type. This is so you can go into details and make calls, like propose, dispute, etc to the correct client.